Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -310,25 +317,30 @@ - (void)lookupDocuments:(const std::vector<DocumentKey> &)keys
 
 - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
                 errorHandler:(FSTVoidErrorBlock)errorHandler {
-  // TODO(mikelehen): We should force a refresh if the previous RPC failed due to an expired token,
-  // but I'm not sure how to detect that right now. http://b/32762461
-  _credentials->GetToken(
-      /*force_refresh=*/false, [self, rpcFactory, errorHandler](util::StatusOr<Token> result) {
-        [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
-          if (!result.ok()) {
-            errorHandler(util::MakeNSError(result.status()));
-          } else {
-            GRPCProtoCall *rpc = rpcFactory();
-            const Token &token = result.ValueOrDie();
-            [FSTDatastore
-                prepareHeadersForRPC:rpc
-                          databaseID:&self.databaseInfo->database_id()
-                               token:(token.user().is_authenticated() ? token.token()
-                                                                      : absl::string_view())];
-            [rpc start];
-          }
-        }];
-      });
+  _credentials->GetToken(_forceTokenRefresh, [self, rpcFactory,
+                                              errorHandler](util::StatusOr<Token> result) {
+    [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
+      _forceTokenRefresh = NO;
+      if (!result.ok()) {
+        NSError *error = util::MakeNSError(result.status());
+        if (error != nil && _lastError != nil &&
+            error.code == FIRFirestoreErrorCodeUnauthenticated &&
```

Comment:
Hrm...  I think it's impossible (or very unlikely) that we get an unauthenticated error from getToken().  We instead expect it from our RPC's to the backend...  so somehow this needs to be moved to https://github.com/firebase/firebase-ios-sdk/blob/ce61f4e0e654e7a16dad023e1b8df32449d5a00f/Firestore/Source/Remote/FSTDatastore.mm#L242 and https://github.com/firebase/firebase-ios-sdk/blob/ce61f4e0e654e7a16dad023e1b8df32449d5a00f/Firestore/Source/Remote/FSTDatastore.mm#L275

Reply:
Done.